keras_nlp/utils/text_generation.py

```diff
@@ -109,16 +153,106 @@ def token_probability_fn(inputs):
         i += 1
 
     if end_token_id is not None:
-        # Mask out tokens after `end_token_id` is encountered.
-        # Find index of first end_token_id.
-        end_indices = tf.math.argmax(prompt == end_token_id, -1)
-        # Use max_length if no `end_token_id` is found.
-        end_indices = tf.where(end_indices == 0, max_length, end_indices)
-        # Build a mask including end_token and replace tokens after end_token
-        # with `pad_token_id`.
-        valid_indices = tf.sequence_mask(end_indices + 1, maxlen=max_length)
-        prompt = tf.where(valid_indices, prompt, pad_token_id)
+        prompt = _mask_tokens_after_end_token(
+            prompt, max_length, end_token_id, pad_token_id
+        )
+
+    if input_is_1d:
+        return tf.squeeze(prompt)
+    return prompt
+
+
+def random_sampling(
```

Comment:
We need to make sure we have naming uniformity. Do we want

`greedy_search`, `random_search`, `top_k_search`? or `greedy_sampling`, `random_sampling`, top_k_sampling`?

Or something else?

Reply:
The way I was going to name it was greedy_search, beam_search, random_sampling, top_k_sampling and top_p_sampling, because the latter three uses probabilistic sampling techniques. But if uniformly calling them search is better, I'll change it to that.

Reply:
I would prefer a name `random_sampling_search`. @mattdangerw Matt, what do you think?